wip: working on a 2.x API update

jcreekmore opened the pull request:
The 2.x API update will support being able to retrieve headers from Pem files as mentioned in #36 in addition to adding `no_std` support for the crate.

cardoe commented:
So first inclination on the first commit is that it should also implement TryFrom for a slice of u8’s.

jcreekmore commented:
> So first inclination on the first commit is that it should also implement TryFrom for a slice of u8’s. 
> 

Added that and squashed the formatting fixes in, along with squashing the two no_std commits together. Can you re-review at some point?

cardoe commented on .github/workflows/ci.yml:
```diff
@@ -18,3 +18,7 @@ jobs:
+      # For no_std we use check instead of test because
+      # std is required for the test suite to run.
+      - run: cargo clippy --no-default-features -- --deny warnings
+      - run: cargo check --no-default-features
```
Not really important for this but you could use a matrix here to run things in parallel
